src/contexts/ZoneBrowserContext.tsx

```diff
@@ -34,77 +38,103 @@ export const ZoneBrowserContextProvider = ({
-        }
+  function createZonesAndFileSharePathsMap(
+    newZonesAndFileSharePathsMap: ZonesAndFileSharePathsMap,
+    item: FileSharePath
+  ) {
+    // Zones first
+    // If the zone doesn't exist in the map, create it
+    if (!newZonesAndFileSharePathsMap[`zone_${item.zone}`]) {
+      newZonesAndFileSharePathsMap[`zone_${item.zone}`] = {
+        name: item.zone,
+        fileSharePaths: []
+      } as Zone;
+    }
+    // If/once zone exists, check if file share path exists in the zone
+    // And update zone accordingly
+    if (newZonesAndFileSharePathsMap[`zone_${item.zone}`]) {
+      const existingZone = newZonesAndFileSharePathsMap[
+        `zone_${item.zone}`
+      ] as Zone;
+      if (
+        !existingZone.fileSharePaths.some(
+          (existingItem: FileSharePath) => existingItem.name === item.name
+        )
+      ) {
+        existingZone.fileSharePaths.push(item);
+        newZonesAndFileSharePathsMap[`zone_${item.zone}`] = existingZone;
+      }
+    }
 
-        // Store the entire FileSharePathItem object instead of just a string path
-        if (
-          !unsortedPaths[item.zone].some(
-            existingItem => existingItem.name === item.name
-          )
-        ) {
-          unsortedPaths[item.zone].push(item);
-        }
-      });
+    // Then add file share paths to the map
+    if (!newZonesAndFileSharePathsMap[`fsp_${item.name}`]) {
+      newZonesAndFileSharePathsMap[`fsp_${item.name}`] = item;
+    }
+  }
 
-      // Sort the items within each zone alphabetically by name
-      Object.keys(unsortedPaths).forEach(zone => {
-        unsortedPaths[zone].sort((a, b) => a.name.localeCompare(b.name));
+  async function updateZonesAndFileSharePathsMap() {
+    let rawData: { paths: FileSharePath[] } = { paths: [] };
+    try {
+      rawData = await getZones();
+      const newZonesAndFileSharePathsMap: Record<string, FileSharePath | Zone> =
+        {};
+      rawData.paths.forEach(item => {
+        createZonesAndFileSharePathsMap(newZonesAndFileSharePathsMap, item);
       });
-
-      // Create a new object with alphabetically sorted zone keys
-      const sortedPaths: ZonesAndFileSharePaths = {};
-      Object.keys(unsortedPaths)
-        .sort()
-        .forEach(zone => {
-          sortedPaths[zone] = unsortedPaths[zone];
-        });
-
-      setZonesAndFileSharePaths(sortedPaths);
+      setZonesAndFileSharePathsMap(newZonesAndFileSharePathsMap);
+      console.log(newZonesAndFileSharePathsMap);
     } catch (error: unknown) {
       if (error instanceof Error) {
         console.error(error.message);
       } else {
-        console.error('An unknown error occurred');
+        console.error('An unknown error occurred when fetching zones');
       }
     }
   }
+
+  // When app first loads, fetch file share paths
+  // and create a map of zones and file share paths
+  React.useEffect(() => {
+    if (Object.keys(zonesAndFileSharePathsMap).length === 0) {
```

Comment:
Not sure I understand the purpose of this `if`. I guess this only loads the data if the map is empty, which means only at the first page load? Wouldn't this prevent new data from being loaded when we navigate between pages or other refresh the view?